pkg/utils/test/matchers/matchers.go

```diff
@@ -146,17 +140,33 @@ func NewManagedResourceContainsObjectsMatcher(cl client.Client) func(...client.O
 // It is expected that the data keys of referenced secret(s) follow the semantics of `managedresources.Registry`.
 func NewManagedResourceConsistOfObjectsMatcher(cl client.Client) func(...client.Object) types.GomegaMatcher {
 	return func(objs ...client.Object) types.GomegaMatcher {
-		expectedObjToSerialized := make(map[client.Object]string)
-		for _, o := range objs {
-			obj := o
-			expectedObjToSerialized[obj] = testruntime.Serialize(obj, cl.Scheme())
+		return &managedResourceObjectsMatcher{
+			ctx:               context.Background(),
+			cl:                cl,
```

Comment:
Nit: We typically use `c client.Client` instead of `cl`. In the struct, you could call it `client`.